webservices/server-integration/src/main/java/org/jboss/as/webservices/config/ServerConfigImpl.java

```diff
@@ -194,7 +194,7 @@ public Integer getVirtualHostPort(String hostname, boolean securePort) {
         Host undertowHost = getUndertowHost(hostInfo);
         if (undertowHost != null && !undertowHost.getServer().getListeners().isEmpty()) {
             if (!securePort) {
-                return undertowHost.getServer().getListeners().get(0).getBinding().getValue().getPort();
+                return undertowHost.getServer().getListeners().get(0).getBinding().getValue().getAbsolutePort();
```

Comment:
The correct way to deal with this is to have some way to specify the listener name you want to reference (defaulting to "default")